frontend/components/MainPage/sections/AccountBalances/AccountBalances.tsx

```diff
@@ -0,0 +1,143 @@
+import { WalletOutlined } from '@ant-design/icons';
+import { Card, Flex, Typography } from 'antd';
+import { useMemo } from 'react';
+import styled from 'styled-components';
+
+import { AddressLink } from '@/components/AddressLink';
+import { InfoBreakdownList } from '@/components/InfoBreakdown';
+import { CustomModal } from '@/components/styled/CustomModal';
+import { COLOR } from '@/constants/colors';
+import { MODAL_WIDTH } from '@/constants/width';
+import { useBalance } from '@/hooks/useBalance';
+import { useWallet } from '@/hooks/useWallet';
+import { balanceFormat } from '@/utils/numberFormatters';
+
+import { Container, infoBreakdownParentStyle } from './styles';
+import { SignerTitle } from './Titles';
+import { YourAgentWallet } from './YourAgent';
+
+const { Title, Text } = Typography;
+
+const MainCard = styled(Card)`
+  > .ant-card-body {
+    padding: 16px;
+  }
+`;
+
+const YourWalletDetails = () => {
+  const { masterSafeAddress } = useWallet();
+
+  return (
+    <Flex vertical gap={12}>
+      <WalletOutlined style={{ fontSize: 24, color: COLOR.TEXT_LIGHT }} />
+      <Flex justify="space-between" align="center" className="w-full">
+        <Title level={5} className="m-0">
+          Your wallet
+        </Title>
+        <AddressLink address={masterSafeAddress} />
+      </Flex>
+    </Flex>
+  );
+};
+
+const OlasBalance = () => {
+  const { safeBalance, totalOlasStakedBalance } = useBalance();
+  const olasBalances = useMemo(() => {
+    return [
+      {
+        title: 'Available',
+        value: balanceFormat(safeBalance?.OLAS ?? 0, 2),
```

Comment:
minor, fine to follow up, `balanceFormat` handles `undefined` and will return `--` for values not yet loaded, not need to fallback to `0`, as this is an incorrect reflection of balance

```suggestion
        value: balanceFormat(safeBalance?.OLAS, 2),
```